pkg/util/dynamichelper/dynamichelper.go

```diff
@@ -438,65 +439,83 @@ func mergeGK(old, new kruntime.Object) (kruntime.Object, bool, string, error) {
-				*expect.Webhooks[i].TimeoutSeconds != *new.Webhooks[i].TimeoutSeconds {
-				*expect.Webhooks[i].TimeoutSeconds = *new.Webhooks[i].TimeoutSeconds
-				changed = true
-			}
-		}
+		changed = handleMutatingWebhook(new, expect)
 	}
-
 	var diff string
 	if _, ok := expect.(*corev1.Secret); !ok { // Don't show a diff if kind is Secret
 		diff = cmp.Diff(old, expect)
 	}
-
 	return expect, changed, diff, nil
 }
 
-func cmpAndCopy(srcP, dstP *corev1.ResourceList) bool {
-	src, dst := *srcP, *dstP
+func handleDeployment(new, expect kruntime.Object) bool {
+	changed := false
+	newDeployment, expectDeployment := new.(*appsv1.Deployment), expect.(*appsv1.Deployment)
+	for i := range expectDeployment.Spec.Template.Spec.Containers {
+		ec := &expectDeployment.Spec.Template.Spec.Containers[i]
+		nc := &newDeployment.Spec.Template.Spec.Containers[i]
+		if ec.Image != nc.Image {
+			ec.Image = nc.Image
+			changed = true
+		}
+		if cmpAndCopy(&nc.Resources.Limits, &ec.Resources.Limits) {
+			changed = true
+		}
+		if cmpAndCopy(&nc.Resources.Requests, &ec.Resources.Requests) {
+			changed = true
+		}
+	}
+	return changed
+}
+
+func handleValidatingWebhook(new, expect kruntime.Object) bool {
+	changed := false
+	newWebhook := new.(*admissionregistrationv1.ValidatingWebhookConfiguration)
+	expectWebhook := expect.(*admissionregistrationv1.ValidatingWebhookConfiguration)
+	for i := range expectWebhook.Webhooks {
+		if expectWebhook.Webhooks[i].FailurePolicy != nil &&
+			newWebhook.Webhooks[i].FailurePolicy != nil &&
+			*expectWebhook.Webhooks[i].FailurePolicy != *newWebhook.Webhooks[i].FailurePolicy {
+			*expectWebhook.Webhooks[i].FailurePolicy = *newWebhook.Webhooks[i].FailurePolicy
+			changed = true
+		}
+		if expectWebhook.Webhooks[i].TimeoutSeconds != nil &&
+			newWebhook.Webhooks[i].TimeoutSeconds != nil &&
+			*expectWebhook.Webhooks[i].TimeoutSeconds != *newWebhook.Webhooks[i].TimeoutSeconds {
+			*expectWebhook.Webhooks[i].TimeoutSeconds = *newWebhook.Webhooks[i].TimeoutSeconds
+			changed = true
+		}
+	}
+	return changed
+}
+
+func handleMutatingWebhook(new, expect kruntime.Object) bool {
+	changed := false
+	newWebhook := new.(*admissionregistrationv1.MutatingWebhookConfiguration)
+	expectWebhook := expect.(*admissionregistrationv1.MutatingWebhookConfiguration)
+	for i := range expectWebhook.Webhooks {
+		if expectWebhook.Webhooks[i].FailurePolicy != nil &&
+			newWebhook.Webhooks[i].FailurePolicy != nil &&
+			*expectWebhook.Webhooks[i].FailurePolicy != *newWebhook.Webhooks[i].FailurePolicy {
+			*expectWebhook.Webhooks[i].FailurePolicy = *newWebhook.Webhooks[i].FailurePolicy
+			changed = true
+		}
+		if expectWebhook.Webhooks[i].TimeoutSeconds != nil &&
+			newWebhook.Webhooks[i].TimeoutSeconds != nil &&
+			*expectWebhook.Webhooks[i].TimeoutSeconds != *newWebhook.Webhooks[i].TimeoutSeconds {
+			*expectWebhook.Webhooks[i].TimeoutSeconds = *newWebhook.Webhooks[i].TimeoutSeconds
+			changed = true
+		}
+	}
+	return changed
+}
```

Comment:
```suggestion
func handleMutatingWebhook(new, expect kruntime.Object) bool {
	newWebhook := new.(*admissionregistrationv1.MutatingWebhookConfiguration)
	expectWebhook := expect.(*admissionregistrationv1.MutatingWebhookConfiguration)
	for i := range expectWebhook.Webhooks {
		if expectWebhook.Webhooks[i].FailurePolicy != nil &&
			newWebhook.Webhooks[i].FailurePolicy != nil &&
			*expectWebhook.Webhooks[i].FailurePolicy != *newWebhook.Webhooks[i].FailurePolicy {
			*expectWebhook.Webhooks[i].FailurePolicy = *newWebhook.Webhooks[i].FailurePolicy
			return true
		}
		if expectWebhook.Webhooks[i].TimeoutSeconds != nil &&
			newWebhook.Webhooks[i].TimeoutSeconds != nil &&
			*expectWebhook.Webhooks[i].TimeoutSeconds != *newWebhook.Webhooks[i].TimeoutSeconds {
			*expectWebhook.Webhooks[i].TimeoutSeconds = *newWebhook.Webhooks[i].TimeoutSeconds
			return true
		}
	}
	return false
}
```

Reply:
also maybe the naming isn't optimal, what do you think of `mutatingHookHasChanged` or something?

Reply:
**hasChanged** is enough, combined with the func name and context, otherwise it is too long.

Reply:
cannot return in the middle, the loop has to be finished, otherwise some changes may be lost.